cpp/bindings/constraints.cpp

```diff
@@ -0,0 +1,112 @@
+#ifdef PINOCCHIO_WITH_PYTHON_INTERFACE
+#include <boost/python.hpp>
+#endif // PINOCCHIO_WITH_PYTHON_INTERFACE
+#endif // BUILD_WITH_PINOCCHIO
+
+namespace toppra {
+namespace python {
+
+using namespace constraint;
+
+void exposeConstraints(py::module m)
+{
+  py::enum_<DiscretizationType>(m, "DiscretizationType")
+      .value("Collocation", DiscretizationType::Collocation)
+      .value("Interpolation", DiscretizationType::Interpolation)
+      .export_values();
+
+  // Abstract class must be binded for derived classes to work
+  py::class_<LinearConstraint, LinearConstraintPtr>(m, "LinearConstraint")
+      .def_property_readonly("nbConstraints", &LinearConstraint::nbConstraints)
+      .def_property_readonly("nbVariables", &LinearConstraint::nbVariables)
+      .def_property_readonly("hasLinearInequalities",
+                             &LinearConstraint::hasLinearInequalities)
+      .def_property_readonly("hasUbounds", &LinearConstraint::hasUbounds)
+      .def_property_readonly("hasXbounds", &LinearConstraint::hasXbounds)
+      .def_property("discretizationType",
+                    (DiscretizationType(LinearConstraint::*)() const) &
+                        LinearConstraint::discretizationType,
+                    (void (LinearConstraint::*)(DiscretizationType)) &
+                        LinearConstraint::discretizationType)
+
+      .def ("computeParams", [](LinearConstraint& constraint, const GeometricPath& path, const Vector& gridpoints) -> py::tuple {
+            Vectors a, b, c, g;
+            Matrices F;
+            Bounds ub, xb;
+            constraint.computeParams(path, gridpoints, a, b, c, F, g, ub, xb);
+            py::list list;
+            py::none none;
+            if (constraint.hasLinearInequalities()) {
+              list.append(a);
+              list.append(b);
+              list.append(c);
+              list.append(F);
+              list.append(g);
+            } else
+              for (int i = 0; i < 5; ++i) list.append(none);
+            if (constraint.hasUbounds()) list.append(ub);
+            else list.append(none);
+            if (constraint.hasXbounds()) list.append(xb);
+            else list.append(none);
+            return list;
+          }
+        );
+    ;
+  py::class_<LinearJointVelocity,
+    std::shared_ptr<LinearJointVelocity>,
+    LinearConstraint>(
+      m, "LinearJointVelocity")
+      .def(py::init<const Vector &, const Vector &>())
+      ;
+  py::class_<LinearJointAcceleration,
+    std::shared_ptr<LinearJointAcceleration>,
+    LinearConstraint>(
+      m, "LinearJointAcceleration")
+      .def(py::init<const Vector &, const Vector &>())
+      ;
+
+  py::class_<JointTorque,
+    std::shared_ptr<JointTorque>,
+    LinearConstraint>(
+      m, "JointTorque");
+
+  {
+    auto mod_jointTorque = m.def_submodule("jointTorque");
+#ifdef BUILD_WITH_PINOCCHIO
+    using jointTorque::Pinocchio;
+    py::module::import("pinocchio");
```

Comment:
@jmirabel  This line requires pinocchio to be available in the python site-packages when the user only wants to use the Pinocchio C++ library. I try to comment out and it doesn't seem to cause any issue.  Do you really need it in this case? If no I will make the below change.

`
#ifdef PINOCCHIO_WITH_PYTHON_INTERFACE
    py::module::import("pinocchio");
#endif
`

Reply:
You are right. Your change is correct.